tests/mlflow/tensorflow/test_tensorflow_model_export.py

```diff
@@ -55,32 +81,126 @@ def test_log_saved_model(self):
+            # Creating the input training function required.
+            trainingFeatures = {}
+
+            for i in df:
+                trainingFeatures[i] = df[i].values
+
+            input_train = tf.estimator.inputs.numpy_input_fn(trainingFeatures, 
+                                                            y_train.values, 
+                                                            shuffle=False, 
+                                                            batch_size=1)
+
+            # Creating the feature columns required for the DNNRegressor.
+            body_style_vocab = ["hardtop", "wagon", "sedan", "hatchback", "convertible"]
+            body_style = tf.feature_column.categorical_column_with_vocabulary_list(
+                key="body-style", vocabulary_list=body_style_vocab)
+            feature_columns = [
+            tf.feature_column.numeric_column(key="curb-weight"),
+            tf.feature_column.numeric_column(key="highway-mpg"),
+            # Since this is a DNN model, convert categorical columns from sparse
+            # to dense.
+            # Wrap them in an `indicator_column` to create a
+            # one-hot vector from the input.
+            tf.feature_column.indicator_column(body_style),]
+
+            # Build a DNNRegressor, with 2x20-unit hidden layers, with the feature columns
+            # defined above as input.
+            model = tf.estimator.DNNRegressor(
+                hidden_units=[20, 20], feature_columns=feature_columns)
+
+            # Training the model.
+            model.train(input_fn=input_train, steps=100)
+            # Saving the model's prediction on the training data.
+            model_predict = model.predict(input_train)
+            # Setting the logging such that it is in the temp folder and deleted after the test.
+            old_tracking_dir = tracking.get_tracking_uri()
+            tracking_dir = os.path.abspath(tmp.path("mlruns"))
+            tracking.set_tracking_uri("file://%s" % tracking_dir)
+            tracking.start_run()
+            try:
+                # Creating dict of features names (str) to placeholders (tensors)
+                feature_spec = {}
+                size = len(y_train.values)
+                feature_spec["body-style"] = tf.placeholder("string", 
+                                                            name="body-style", 
+                                                            shape=[size])
+                feature_spec["curb-weight"] = tf.placeholder("float", 
+                                                            name="curb-weight", 
+                                                            shape=[size])
+                feature_spec["highway-mpg"] = tf.placeholder("float", 
+                                                            name="highway-mpg", 
+                                                            shape=[size])
+
+                results = self.helper(feature_spec, tmp, model, model_predict, df)
```

Comment:
Thanks for refactoring this, always good to reduce code duplication.

It seems like `self.helper`'s main role is to export a fitted model, load it back as a pyfunc, & compute its predictions on a dataset. Maybe we can trim it down to do just that / that way we don't need to pass `model_predict`? In the same vein we can just return the pyfunc's predictions on the dataset as opposed to a tuple of `(estimator_predictions, pyfunc_predictions)`, since we already have the estimator predictions.